Makefile

```diff
@@ -12,3 +12,6 @@ generate:
 
 update-licenses:
 	go run github.com/elastic/go-licenser@v0.4.1 .
+
+fieldalignment:
+	go run golang.org/x/tools/go/analysis/passes/fieldalignment/cmd/fieldalignment@v0.4.0 -test=false $(shell go list ./... | grep -v modeldecoder/generator | grep -v test)
```

Comment:
Should it also be part of the `generate` command?

Reply:
I'm hesitant to add it to fmt since `fieldalignment` removes the godoc struct comments when it reorders the fields. I haven't found a way to make it respect those and it seems to be a limitation on the tool.

Reply:
ok, so it would rather be a manual one-off task to run occasionally?

Reply:
I think as we add fields to structs some will need to be realigned. The process I have followed has been:

1. (manual) Align from bigger to smaller (pointers, slices, maps, etc generally go first, then strings, float, int, bool).
2. If that still doesn't fully align and make it compliant with `fieldalignment`, then run `fieldalignment -test=false -fix` which will rewrite the structures
3. Copy the rewritten structures into an unsaved file
4. Revert the change and sort the fields like field alignment would like them with the corresponding comments.

It's not great, but it's the best way I've found to preserve the comments.

Reply:
There's an open issue on the Go repo for it: https://github.com/golang/go/issues/43735

Reply:
generally sounds ok to me; I tried running `make generate` (after rebasing onto current main) and the generator logic is broken on this brunch (but works in main). Please take a look at what's the issue with it before moving forward.

Reply:
Fixed `make generate` in [c066583](https://github.com/elastic/apm-data/pull/4/commits/c0665832493d00fafb1f3ba8561b6994a84499ee)